addons/payment_adyen/utils.py

```diff
@@ -53,15 +53,21 @@ def format_partner_address(partner):
     """
     STREET_FORMAT = '%(street_number)s/%(street_number2)s %(street_name)s'
     street_data = split_street_with_params(partner.street, STREET_FORMAT)
-    return {
+    state_or_province = partner.state_id.code
+    if not state_or_province and partner.country_id.code in ['CA', 'US']:
+        raise ValidationError(_("Please fill out the address state."))
+    address = {
         'city': partner.city,
         'country': partner.country_id.code or 'ZZ',  # 'ZZ' if the country is not known.
-        'stateOrProvince': partner.state_id.code,
+        'stateOrProvince': state_or_province or '',
         'postalCode': partner.zip,
         # Fill in the address fields if the format is supported, or fallback to the raw address.
         'street': street_data.get('street_name', partner.street),
         'houseNumberOrName': street_data.get('street_number'),
     }
+    if False in address.values():
+        raise ValidationError(_("Please fill out the address info"))
```

Comment:
```
    if False in address.values() or (partner.country_id.code in ['CA', 'US'] and not state_or_province):
        raise ValidationError(_("Please complete your address details"))
```
Do you think this is acceptable solution? @AntoineVDV

Reply:
Hm, because I can't come up with anything other than this or two checks.

Reply:
Since the two warnings serve different purposes (prevent empty state details for US and CA and prevent `False` being sent to Adyen for any field), we could get rid of the second one by simply appending `or ''` to all fields.

Reply:
But from what I understand it's not what Adyen expects. https://docs.adyen.com/risk-management/avs-checks/ It says that all values of address are required.

Reply:
Oh ok I missed that! Then, one plain dumb check that explicitly loops over the dict keys and raises an error. It does not raise when the key is `stateOrProvince` and the country is not in [US, CA, UK] (notice that this pages mentions UK still)

Reply:
Checked, Adyen is refusing if other values are set to ''.

Reply:
I'm not sure if I understand you correctly, is that what you mean?
```suggestion
    for line in address:
        if address[line] == False:
            raise ValidationError(_("Please complete your address details"))
        if line == 'stateOrProvince' and (partner.country_id.code in ['CA', 'US'] and not state_or_province):
            raise ValidationError(_("Please complete your address details"))
```

Reply:
Yes :)

Use `for key, value in address.items()` though.